jdaviz/utils.py

```diff
@@ -308,8 +309,13 @@ def flux_conversion(spec, values, original_units, target_units):
 
     Returns
     -------
-    Flux values in the target units.
+    result : float array
+        Flux values in the target units.
     """
+    # we set surface brightness choices and selection before flux, which can
+    # cause a dimensionless translation attempt at instantiation
+    if not target_units:
+        target_units = original_units
```

Comment:
in this case, can we just return the input array?

Reply:
I think I did have this as `return values` before. I don't remember exactly why I forced it to continue through the flux_conversion function (maybe I thought it had to do with a test failure). I just reran tests locally and tested out the UI with the change, tests are passing and the behavior is as it should be so I will make the switch.